packages/core/src/shared/telemetry/spans.ts

```diff
@@ -364,6 +375,34 @@ export class TelemetryTracer extends TelemetryBase {
         }
     }
 
+    /**
+     * Trace represents a higher level event that you want to link with any other spans. It generates
+     * a traceId which is added to all subsequent calls, allowing you to query for all related events.
+     *
+     * Functionally, it is similiar to run. The only difference is the traceId gets added to subsequent telemetry calls.
+     *
+     * See: https://opentelemetry.io/docs/concepts/signals/traces/ to better understand the motivation behind traces
+     *
+     * @param name The name of the trace
+     * @param fn the callback function
+     */
+    private trace<T, U extends MetricName>(name: U, fn: (span: Metric<MetricShapes[U]>) => T): T {
+        return this.runRoot(() => {
+            /**
+             * Make sure we aren't starting a trace inside of a trace. traceId's have to be unique for every
+             * flow, otherwise we can't really tell what the trace belongs to
+             */
+            if (this.attributes?.traceId) {
+                throw new Error('Cannot trace an already tracing telemetry event')
```

Comment:
related to my comment about making this in `run` by default.

Instead of throwing an error we can just do nothing/use the already existing traceId. It can be extremely hard to tell the stack trace of a call in advance- we may be already tracing upstream.